spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskJobService.java

```diff
@@ -254,10 +254,12 @@ public void restartJobExecution(long jobExecutionId, String schemaTarget) throws
 	 * identifying job parameters not in the original task execution arguments.
 	 */
 	private List<String> restartExecutionArgs(List<String> taskExecutionArgs, JobParameters jobParameters, String schemaTarget) {
+		if(schemaTarget.equals(SchemaVersionTarget.createDefault(AppBootSchemaVersion.BOOT2).getName())) {
```

Comment:
Would it be safer to invert this and ensure that the `schemaTarget` is boot3?

Reply:
I'll add a boot3 comment.  Because we will be removing the Boot2 code.   But at this moment, until we remove it, I wanted to let folks this was a no no.

Reply:
Agreed. It may be a wider net to say "If this is not boot3 BOOM" rather than "if this is boot2 BOOM"... however I think the inputs are constrained to one or the other so it really does not matter. Especially since we will be removing it. Disregard. Thx. :)